failing tests with pandas  2.1

### What happened?

See  https://github.com/pydata/xarray/pull/8101

```
FAILED xarray/tests/test_missing.py::test_interpolate_pd_compat - ValueError: 'fill_value' is not a valid keyword for DataFrame.interpolate
FAILED xarray/tests/test_missing.py::test_interpolate_pd_compat_non_uniform_index - ValueError: 'fill_value' is not a valid keyword for DataFrame.interpolate
```

and this doctest
```
FAILED xarray/core/dataarray.py::xarray.core.dataarray.DataArray.to_unstacked_dataset
```
@pydata/xarray can someone take a look please?

> ```
> FAILED xarray/tests/test_missing.py::test_interpolate_pd_compat - ValueError: 'fill_value' is not a valid keyword for DataFrame.interpolate
> FAILED xarray/tests/test_missing.py::test_interpolate_pd_compat_non_uniform_index - ValueError: 'fill_value' is not a valid keyword for DataFrame.interpolate
> ```

``fill_value`` - kwarg was removed upstream and raises error now -> https://github.com/pandas-dev/pandas/pull/53962

> FAILED xarray/core/dataarray.py::xarray.core.dataarray.DataArray.to_unstacked_dataset

The doctest works for me locally, investigating package differences in the environment now. I

--- a/xarray/tests/test_missing.py
+++ b/xarray/tests/test_missing.py
@@ -104,7 +104,8 @@ def test_interpolate_pd_compat():
         for dim in ["time", "x"]:
             actual = da.interpolate_na(method=method, dim=dim, fill_value=np.nan)
             expected = df.interpolate(
-                method=method, axis=da.get_axis_num(dim), fill_value=(np.nan, np.nan)
+                method=method,
+                axis=da.get_axis_num(dim),
             )
             # Note, Pandas does some odd things with the left/right fill_value
             # for the linear methods. This next line inforces the xarray
@@ -140,7 +141,8 @@ def test_interpolate_pd_compat_non_uniform_index():
                 method="linear", dim=dim, use_coordinate=True, fill_value=np.nan
             )
             expected = df.interpolate(
-                method=method, axis=da.get_axis_num(dim), fill_value=np.nan
+                method=method,
+                axis=da.get_axis_num(dim),
             )
 
             # Note, Pandas does some odd things with the left/right fill_value

